### PR TITLE
Rental service implementation

### PR DESCRIPTION
**Pull request by CodeWebWeaver**
This request includes:
- RentalServiceImpl logic
- RentalDto fix (boolean -> Boolean isActive)
- Rental Specification search
- removed @Repository annotation from UserRepository interface
- moved PaymentServiceImpl to impl folder

Ready for conversation!

### CODE REVIEW

**Review by holsoni — APPROVED** [review]
LGTM!

File: src/main/java/com/jvavateam/carsharingapp/repository/SpecificationProvider.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+import com.jvavateam.carsharingapp.model.Rental;
+import org.springframework.data.jpa.domain.Specification;
+
+public interface SpecificationProvider<T> {
+    String getKey();
+
+    Specification<Rental> getSpecification(Boolean params);
+
+    Specification<Rental> getSpecification(Long params);
+}
```
Comment (Dimagaa) [inline src/main/java/com/jvavateam/carsharingapp/repository/SpecificationProvider.java]: Can we make `getSpecification()` method generic? What if we want to search by rental date?

File: src/main/java/com/jvavateam/carsharingapp/service/impl/RentalServiceImpl.java [hunk 2]
```diff
@@ -3,32 +3,115 @@
...
+        Car actualCar = carRepository.findById(decreasingCarId)
+                .orElseThrow(() ->
+                        new EntityNotFoundException("Can`t find car with id: " + decreasingCarId));
+        carForIncreasing.setInventory(actualCar.getInventory() + 1);
+        carRepository.save(carForIncreasing);
+    }
+
+    private void decreaseCarInventory(Long carIdForDecreasing) {
+        Car actualCar = carRepository.findById(carIdForDecreasing)
+                .orElseThrow(() -> new EntityNotFoundException("Car with provided id not found"));
```
Comment (Dimagaa) [inline src/main/java/com/jvavateam/carsharingapp/service/impl/RentalServiceImpl.java]: I think we should use Car Service instead Car Repository

File: src/main/java/com/jvavateam/carsharingapp/service/impl/RentalServiceImpl.java [hunk 3]
```diff
@@ -3,32 +3,115 @@
...
+    }
+
+    private boolean isManager(User user) {
+        return user.getRoles()
+                .stream()
+                .anyMatch(role -> role.getName().equals(Role.RoleName.MANAGER));
+    }
+
+    private void checkUserAccess(User user) {
+        Long currentUserId = userRepository.getCurrentUser().getId();
```
Comment (Dimagaa) [inline src/main/java/com/jvavateam/carsharingapp/service/impl/RentalServiceImpl.java]: We can check it in Rental Repository

File: src/main/java/com/jvavateam/carsharingapp/service/impl/RentalServiceImpl.java [hunk 4]
```diff
@@ -3,32 +3,115 @@
...
+        checkCarInventory(actualCar);
+        actualCar.setInventory(actualCar.getInventory() - 1);
+        carRepository.save(actualCar);
+    }
+
+    private boolean isManager(User user) {
+        return user.getRoles()
+                .stream()
+                .anyMatch(role -> role.getName().equals(Role.RoleName.MANAGER));
+    }
```
Comment (Dimagaa) [inline src/main/java/com/jvavateam/carsharingapp/service/impl/RentalServiceImpl.java]: Lets try do it in repository

**Review by Dimagaa — CHANGES_REQUESTED** [review]
Great Job! Lets improve it!

**Pull request closed without merging**